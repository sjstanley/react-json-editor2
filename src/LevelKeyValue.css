.LevelKeyValue {
  /*border: 2px solid black;*/
  /*text-align: center;*/
  background: #f5f5f5;
  /*color: #333;*/
  /*display: none;*/
  margin-left: 10px;
  padding: 10px;
  border-left: 1px dotted #ddd;
}

.compoundAttr > .LevelKeyValue {
	display: block;
}

.objectAttr {
	float: left;
}

.attrName {
  font-style: italic;
}

.attrValue {
  margin-left: 5px;
}