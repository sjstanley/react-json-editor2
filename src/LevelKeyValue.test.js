import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import LevelKeyValue from './LevelKeyValue';

describe(LevelKeyValue, () => {
	const name = 'Person';
	const mockRemoveGreeting = jest.fn();
	const component = shallow(
		<LevelKeyValue name={name} removeGreeting={mockRemoveGreeting}/>
	);

	it('renders and matches our snapshot', () => {
		const component = renderer.create(
			<LevelKeyValue name="Person" />
		);
		const tree = component.toJSON();
		expect(tree).toMatchSnapshot();
	});

	it('contains the supplied name', () => {
		expect(component.text()).toContain(name);
	});

	it('calls the passed in removeGreeting function when remove button is clicked', () => {
		component.find('button.remove').simulate('click');
		expect(mockRemoveGreeting).toBeCalled();
	});

});
