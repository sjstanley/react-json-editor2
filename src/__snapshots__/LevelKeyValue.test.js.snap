exports[`class LevelKeyValue extends /* istanbul ignore next */_react.Component {

	constructor(props) {
		super(props);
		this.state = { greeting: 'Hello', modifiedValue: '' };
		this.removeGreeting = this.removeGreeting.bind(this);
		this.handleUpdate = this.handleUpdate.bind(this);
		this.handleUpdateClick = this.handleUpdateClick.bind(this);

	}

	removeGreeting() {
		this.props.removeGreeting(this.props.name);
	}

	handleUpdate(event) {
		this.setState({ greeting: 'Hello', modifiedValue: event.target.value });
	}
	handleUpdateClick(event) {
		this.props.updateGreeting(this.props.name, this.state.modifiedValue);
		this.setState({ greeting: 'Hello', modifiedValue: '' });
	}


	render() {
		return (
			/* istanbul ignore next */_react2.default.createElement( /* istanbul ignore next */'div', /* istanbul ignore next */{ className: 'LevelKeyValue', __source: { fileName: _jsxFileName, lineNumber: 30 }, __self: this }, /* istanbul ignore next */'Item ',
				this.props.index, /* istanbul ignore next */' : ', this.props.name,
				/* istanbul ignore next */_react2.default.createElement( /* istanbul ignore next */'br', /* istanbul ignore next */{ __source: { fileName: _jsxFileName, lineNumber: 32 }, __self: this }),
				/* istanbul ignore next */_react2.default.createElement( /* istanbul ignore next */'input', /* istanbul ignore next */{
					type: 'text',
					onChange: this.handleUpdate,
					value: this.state.modifiedValue, __source: { fileName: _jsxFileName, lineNumber: 33 }, __self: this }),

				/* istanbul ignore next */_react2.default.createElement( /* istanbul ignore next */'button', /* istanbul ignore next */{ className: 'update', onClick: this.handleUpdateClick, __source: { fileName: _jsxFileName, lineNumber: 38 }, __self: this }, /* istanbul ignore next */'Update'),
				/* istanbul ignore next */_react2.default.createElement( /* istanbul ignore next */'button', /* istanbul ignore next */{ className: 'remove', onClick: this.removeGreeting, __source: { fileName: _jsxFileName, lineNumber: 39 }, __self: this }, /* istanbul ignore next */'Remove')));


	}} renders and matches our snapshot 1`] = `
<div
  className="LevelKeyValue">
  Item 
   : 
  Person
  <br />
  <input
    onChange={[Function]}
    type="text"
    value="" />
  <button
    className="update"
    onClick={[Function]}>
    Update
  </button>
  <button
    className="remove"
    onClick={[Function]}>
    Remove
  </button>
</div>
`;
